Objects/odictobject.c

```diff
@@ -1837,38 +1837,17 @@ PyDoc_STRVAR(reduce_doc, "Return state information for pickling");
 static PyObject *
 odictiter_reduce(odictiterobject *di)
 {
-    PyObject *list, *iter;
-
-    list = PyList_New(0);
-    if (!list)
-        return NULL;
+    /* copy the iterator state */
+    odictiterobject tmp = *di;
+    Py_XINCREF(tmp.di_odict);
```

Comment:
This is not enough. The state includes a reference to the result tuple.

Reply:
It looks like the problem wasn't caused by the result tuple.